expyfun/stimuli/_hrtf.py

```diff
@@ -40,29 +39,82 @@ def _get_hrtf(angle, source, fs):
+
+    If interp is True, the function takes the two nearest known HRTFs and uses
+    them to calculate an HRTF between them for any given azimuthal angle
+    between 0 and 90 degrees.
+
+    Interpolation relies on averaging log magnitude and phase in the frequency
+    domain. [1]
+
+    References
+    ----------
+    [1] R.  Martin and K.  McAnally, "Interpolation of Head-Related Transfer
+    Functions", Australian Government Department of Defence: Defence Science
+    and Technology Organization, Melbourne, Victoria, Australia, 2007.
     """
+
     fname = fetch_data_file('hrtf/{0}_{1}.hdf5'.format(source, fs))
     data = read_hdf5(fname)
     angles = data['angles']
     leftward = False
-    read_angle = angle
+    read_angle = float(angle)
     if angle < 0:
         leftward = True
-        read_angle = -angle
-    if read_angle not in angles:
+        read_angle = float(-angle)
+    if read_angle not in angles and not interp:
         raise ValueError('angle "{0}" must be one of +/-{1}'
                          ''.format(angle, list(angles)))
     brir = data['brir']
-    idx = np.where(angles == read_angle)[0]
-    assert len(idx) == 1
-    brir = brir[idx[0]].copy()
+    if read_angle in angles:
+        interp = False
+    if not interp:
+        idx = np.where(angles == read_angle)[0]
+        if len(idx) != 1:
+            raise ValueError('angle "{0}" not uniquely found in angles'
+                             ''.format(angle))
+        brir = brir[idx[0]]
+    else:  # interpolation
+        if source != 'cipic':
+            raise ValueError('source must be ''cipic'' when interp=True')
+
+        # pull in files containing known hrtfs and extract magnitude and phase
+        fname = fetch_data_file('hrtf/pair_cipic_{0}.hdf5'.format(fs))
+        data = read_hdf5(fname)
+        hrtf_amp = data['hrtf_amp']
+        hrtf_phase = data['hrtf_phase']
+        pairs = data['pairs']
+
+        # isolate appropriate pair of amplitude and phase
+        idx = np.searchsorted(angles, read_angle)
+        if idx > len(pairs):
+            raise ValueError('angle magnitude "{0}" must be smaller than "{1}"'
+                             ''.format(read_angle, pairs[-1][-1]))
+        knowns = np.array([angles[idx - 1], angles[idx]])
+        index = np.where(pairs == knowns)[0][0]
+        hrtf_amp = hrtf_amp[index]
+        hrtf_phase = hrtf_phase[index]
+
+        # weighted averages of log magnitude and unwrapped phase
+        step = float(knowns[1] - knowns[0])
+        weights = (step - np.abs(read_angle - knowns)) / step
+        hrtf_amp = np.prod(hrtf_amp ** weights[:, np.newaxis, np.newaxis],
+                           axis=0)
+        hrtf_phase = np.sum(hrtf_phase * weights[:, np.newaxis, np.newaxis],
+                            axis=0)
+
+        # reconstruct hrtf and convert to time domain
+        hrtf = hrtf_amp * np.exp(1j * hrtf_phase)
+        hrtf = _make_sym(hrtf)
+        brir = np.real(np.fft.ifft(hrtf))
```

Comment:
I think you could actually leave the data as is and just do ``np.irfft`` on the first half of them. You can get the length information you need from the data file because it is the full FFT. I have not used ``rfft`` or ``irfft`` before. Eric does that seem right?